src/AutoUI/Core/Metadata/DefaultEntityPropertyListProvider.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using DotVVM.AutoUI.Annotations;
+
+namespace DotVVM.AutoUI.Metadata
+{
+    /// <summary>
+    /// Provides a list of properties for the specified entity.
+    /// </summary>
+    public class DefaultEntityPropertyListProvider : IEntityPropertyListProvider
+    {
+        private readonly IPropertyDisplayMetadataProvider propertyDisplayMetadataProvider;
+        
+        public DefaultEntityPropertyListProvider(IPropertyDisplayMetadataProvider propertyDisplayMetadataProvider)
+        {
+            this.propertyDisplayMetadataProvider = propertyDisplayMetadataProvider;
+        }
+
+        /// <summary>
+        /// Gets a list of properties for the specified entity and view name.
+        /// </summary>
+        public IEnumerable<PropertyDisplayMetadata> GetProperties(Type entityType, IViewContext viewContext)
+        {
+            var metadata = entityType.GetProperties()
```

Comment:
It is unnecessary to sort all the properties and filter after that. Reordering the operations a bit should yield better performance.